src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -436,6 +438,16 @@ module.exports = function<T, P, I, TI, C, CX>(
     } else {
       // Proceed under the assumption that this is a functional component
       workInProgress.tag = FunctionalComponent;
+      if (__DEV__) {
+        warning(
+          !!workInProgress.ref,
+          'Stateless function components cannot be given refs ' +
+          '(See ref "' + workInProgress.ref._stringRef + '" in ' +
+          getComponentName(workInProgress) + ' created by ' +
+          getComponentName(workInProgress.return) + '). ' +
```

Comment:
It seems as though `ReactDebugCurrentFiber.getCurrentFiberStackAddendum()` and `ReactComponentTreeHook.getStackAddendumByID(this._debugID)` return slightly different stacks.

Fiber:
```
      "Warning: Stateless function components cannot be given refs. Attempts to access this ref will fail.
        in StatelessComponent (at ReactStatelessComponent-test.js:156)
        in Parent (at ReactStatelessComponent-test.js:160)"
```

Stack
```
      "Warning: Stateless function components cannot be given refs Attempts to access this ref will fail.
        in Parent (at ReactStatelessComponent-test.js:160)"
```

This is with a render method of:

```
      render() {
        return <div><StatelessComponent name="A" ref="stateless"/></div>;
      }
```

Is this acceptable deviance? If yes should I just make the test loose enough to assert on the commonalities? If not how do you want me to proceed?